src/main/java/core/basesyntax/impl/StorageImpl.java

```diff
@@ -3,17 +3,49 @@
 import core.basesyntax.Storage;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private static final int MAX_ARRAY_LENGTH = 10;
+    private static final int START_INDEX = 0;
+    private static final int NO_MATCH_INDEX = -1;
+    private final K[] keys;
+    private final V[] values;
+    private int size;
+
+    public StorageImpl() {
+        keys = (K[]) new Object[MAX_ARRAY_LENGTH];
+        values = (V[]) new Object[MAX_ARRAY_LENGTH];
+    }
+
     @Override
     public void put(K key, V value) {
+        int index = getIndexForPutting(key);
+        if (index == NO_MATCH_INDEX) {
+            keys[size] = key;
+            values[size] = value;
+            size++;
+        } else {
+            values[index] = value;
+        }
     }
 
     @Override
     public V get(K key) {
-        return null;
+        return getIndexForPutting(key) == NO_MATCH_INDEX ? null
+                : values[getIndexForPutting(key)];
     }
 
     @Override
     public int size() {
-        return -1;
+        return size;
+    }
+
+    private int getIndexForPutting(K key) {
+        for (int i = START_INDEX; i < size; i++) {
+            if ((key == keys[i]) || key != null && key.equals(keys[i])) {
```

Comment:
Based on mentor's review of my homework I have to add one more change request:
You should extract logical condition from "if" to a separate method. Like this: 
```suggestion
           if (keysAreEqual(...) {
```
Don't ask me why :-D